ci/pkl-pipelines/common/shared_resources.pkl

```diff
@@ -2,36 +2,129 @@ import "package://pkg.pkl-lang.org/github.com/alphagov/pkl-concourse-pipeline/pk
 import "package://pkg.pkl-lang.org/github.com/alphagov/pkl-concourse-pipeline/pkl-concourse-pipeline@0.0.1#/TaskConfig.pkl"
 
 class PayGitHubResource extends Pipeline.Resource {
-    type = "git"
-    icon = "github"
+  type = "git"
+  icon = "github"
 }
 
 anonymousConcourseRunnerResource = new TaskConfig.AnonymousResource {
-    type = "registry-image"
-    source = new {
-        ["repository"] = "governmentdigitalservice/pay-concourse-runner"
-        ["tag"] = "latest"
-    }
+  type = "registry-image"
+  source = new {
+    ["repository"] = "governmentdigitalservice/pay-concourse-runner"
+    ["tag"] = "latest"
+  }
 }
 
 payCiGitHubResource: PayGitHubResource = new {
-    name = "pay-ci"
-    source = new Mapping {
-        ["uri"] = "https://github.com/alphagov/pay-ci"
-        ["branch"] = "master"
+  name = "pay-ci"
+  source = new Mapping {
+    ["uri"] = "https://github.com/alphagov/pay-ci"
+    ["branch"] = "master"
+    ["username"] = "alphagov-pay-ci-concourse"
+    ["password"] = "((github-access-token))"
+  }
+}
+
+function payDockerHubResource(resource_name: String, repo: String, tag: String, checkInterval: String): Pipeline.Resource = new {
+  name = resource_name
+  type = "registry-image"
+  icon = "docker"
+  when (checkInterval.length > 0) {
+    check_every = checkInterval
+  }
+  source = new {
+    ["repository"] = repo
+    ["tag"] = tag
+    ["username"] = "((docker-username))"
+    ["password"] = "((docker-access-token))"
+  }
+}
+
+function payGithubResource(resource_name: String, repo: String, tagRegex: String, path: String, repo_branch: String): Pipeline.Resource = new {
```

Comment:
I was going to say path can be a list, but actually looking at what we have it's almost always a single path. The couple that need it to be a list can amend it like:

```
(shared_resources.payGithubResource(...SNIP...)) {
  source {
    ["paths"]  {
       ...SNIP....
    }
}
```

Reply:
Updated

Reply:
Is this what @jfharden was meaning? I took it to mean that defaulting to a single value was fine and that if we needed a list we could override. No?